modin/core/dataframe/pandas/partitioning/partition_manager.py

```diff
@@ -67,12 +72,11 @@ def wait(*args, **kwargs):
             # serially kick off all the deferred computations so that they can
             # all run asynchronously, then wait on all the results.
             [part.drain_call_queue() for part in partitions.flatten()]
-            # need to go through all the values of the map iterator
-            # since `wait` does not return anything, we need to explicitly add
-            # the return `True` value from the lambda
-            # TODO(https://github.com/modin-project/modin/issues/4491): Wait
-            # for all the partitions in parallel.
-            all(map(lambda partition: partition.wait() or True, partitions.flatten()))
+            # The PandasOnRay partition manager invokes ray.get under the hood, which waits
+            # in parallel for all computations to finish
+            PandasOnRayDataframePartitionManager.get_objects_from_partitions(
```

Comment:
`get_objects_from_partitions` will materialize all the partition data in the main process. We should avoid that as much as possible. It's not uncommon on a distributed cluster for the worker node to not have enough memory to hold all the data. Even when running ray on one machine we will have to pay for deserializing and copying all the data from the object store. For ray we want something like `ray.wait([partition._data for partition in partitions], num_returns=len(partitions))`.

Reply:
Ah, I didn't realize that was the difference between `get` and `wait`. Is it correct to say that both methods will wait for all computations to finish, but `wait` doesn't copy the data from the object store?

Also, should we add a separate method to the partition manager to avoid having to access the `partition._data` field from here?

Reply:
> Is it correct to say that both methods will wait for all computations to finish, but wait doesn't copy the data from the object store?

That's the main difference.

> Also, should we add a separate method to the partition manager to avoid having to access the partition._data field from here?

I think so. That would also match `get_objects_from_partitions`, which will have a very similar purpose.